app/backend/approaches/approach.py

```diff
@@ -161,7 +163,16 @@ async def search(
                         reranker_score=document.get("@search.reranker_score"),
                     )
                 )
-        return documents
+
+            qualified_documents = [
```

Comment:
Could you add a unit test in our pytest test suite that checks this logic? You can test the Approach.search() function directly and use a mock of SearchClient.search, we have that mocked in several places in our tests. Let me know if you need further guidance. Thanks!